### ISSUE TITLE
Allow users to add shortcuts to HChooser

### ISSUE THREAD

**Issue by tami5** [Hammerspoon/hammerspoon#3137]
Hey, it would be great if I can inject custom shortcuts to `windowDidBecomeKey`  with callback from lua. Any ideas.

Thanks

**Comment by latenitefilms** [comment]
This might give you some ideas:

https://github.com/CommandPost/CommandPost/blob/develop/src/plugins/core/action/manager/activator.lua

**Comment by tami5** [comment]
@latenitefilms 😍 amazing work.

I'm seeing some mouse handling stuff and I'm like wow. I'll try to look more into it. But wouldn't be just easier if a table would be passed to chooser to setup callbacks?

**Comment by tami5** [comment]
so showCallback needs to be set with setupEventtap? which I should handle key codes there? interesting I'll bite

**Comment by latenitefilms** [comment]
Absolutely, I think your initial idea is great - but just wanted to provide a possible workaround in the meantime, because hs.chooser has historically always been a bit of a beast to change/update/improve.

**Comment by tami5** [comment]
@latenitefilms Thanks so much, I really was struggling with this one, here where I am right now.

```lua
self.chooser:showCallback(function()
    if not self.eventtap then
      self.eventtap = require("hs.eventtap").new({ eventtap.event.types.keyDown }, function(event)
        if event:getFlags():containExactly { "ctrl" } then
          local keycode = event:getKeyCode()
          v("ctrl+{}", keycode)
          print(keycode)
          if keycode == 32 then
            self.chooser:query ""
          elseif keycode == 13 then
            eventtap.keyStroke({ "alt" }, "Delete", 0)
          elseif keycode == 38 then
            eventtap.keyStroke({ "ctrl" }, "n", 0)
          elseif keycode == 40 then
            eventtap.keyStroke({ "ctrl" }, "p", 0)
          end
        end
      end)
    end
    if self.eventtap then
      self.eventtap:start()
    end
  end)
```

I will keep hacking and see how to get a map of shortcuts to `windowDidBecomeKey` ideally I want to have something similar to https://github.com/tami5/Binder.spoon.

I think this is sufficient for simple use case, but my end goal is to dynamically add shortcuts depending on the type of the entry.

**Comment by tami5** [comment]
Okay tried to tackle this one but didn't get anywhere. Objective-c is just so foreign to me
